sormas-backend/src/main/resources/sql/sormas_schema.sql

```diff
@@ -11900,4 +11902,25 @@ UPDATE person_history SET birthcountry_id = NULL WHERE birthcountry_id IS NOT NU
 
 INSERT INTO schema_version (version_number, comment) VALUES (486, 'Hide citizenship and country of birth #9598');
 
+-- 2022-08-11 User roles should have optional link to default user role #9645
+
+CREATE OR REPLACE FUNCTION add_column(in_statement TEXT, in_table TEXT, in_column TEXT, in_schema TEXT DEFAULT 'public') RETURNS BOOLEAN AS $_$
+BEGIN
+    PERFORM * FROM information_schema.columns WHERE table_name = in_table AND column_name = in_column AND table_schema = in_schema;
+    IF FOUND THEN
+        RETURN FALSE;
+    ELSE
+        EXECUTE in_statement;
+        RETURN TRUE;
+    END IF;
+END
+$_$ LANGUAGE plpgsql VOLATILE;
+
+DO $$ BEGIN
+   PERFORM add_column('ALTER TABLE userroles ADD COLUMN linkeddefaultuserrole varchar(255)', 'userroles', 'linkeddefaultuserrole');
+   PERFORM add_column('ALTER TABLE userroles_history ADD COLUMN linkeddefaultuserrole varchar(255)', 'userroles_history', 'linkeddefaultuserrole');
```

Comment:
the function more generic and generate the query using some variables, it also could handle adding the column on the history table if you want the function to be reusable. Otherwise it would be better to delete the function at the end of the migration.

Reply:
i've changed it so that the parameters are used in the statement generation
i would not include the history table modification in it
we don't have a function for the normal addition either that modifies both tables at once, and there are some rare tables that don't have a history one

Reply:
ok